L1Trigger/DTTriggerPhase2/plugins/DTTrigPhase2Prod.cc

```diff
@@ -0,0 +1,719 @@
+  std::vector<metaPrimitive> metaPrimitives;
+  MuonPathPtrs outmpaths;
+  if (algo_ == Standard) {
+    if (debug_)
+      LogDebug("DTTrigPhase2Prod") << "Fitting 1SL ";
+    mpathanalyzer_->run(iEvent, iEventSetup, filteredmuonpaths, metaPrimitives);
+  } else {
+    // implementation for advanced (2SL) grouping, no filter required..
+    if (debug_)
+      LogDebug("DTTrigPhase2Prod") << "Fitting 2SL at once ";
+    mpathanalyzer_->run(iEvent, iEventSetup, muonpaths, outmpaths);
+  }
+
+  if (dump_) {
+    for (unsigned int i = 0; i < outmpaths.size(); i++) {
+      LogDebug("DTTrigPhase2Prod") << iEvent.id().event() << " mp " << i << ": " << outmpaths.at(i)->bxTimeValue()
+                                   << " " << outmpaths.at(i)->horizPos() << " " << outmpaths.at(i)->tanPhi() << " "
+                                   << outmpaths.at(i)->phi() << " " << outmpaths.at(i)->phiB() << " "
+                                   << outmpaths.at(i)->quality() << " " << outmpaths.at(i)->chiSquare() << " ";
+      LogDebug("DTTrigPhase2Prod");
+    }
+    for (unsigned int i = 0; i < metaPrimitives.size(); i++) {
+      LogDebug("DTTrigPhase2Prod") << iEvent.id().event() << " mp " << i << ": ";
+      printmP(metaPrimitives.at(i));
+      LogDebug("DTTrigPhase2Prod");
+    }
+  }
+
+  muonpaths.clear();
+  filteredmuonpaths.clear();
+
+  /////////////////////////////////////
+  //  FILTER SECTIONS:
+  ////////////////////////////////////
+  if (debug_)
+    LogDebug("DTTrigPhase2Prod") << "declaring new vector for filtered" << std::endl;
+
+  std::vector<metaPrimitive> filteredMetaPrimitives;
+  if (algo_ == Standard)
+    mpathqualityenhancer_->run(iEvent, iEventSetup, metaPrimitives, filteredMetaPrimitives);
+
+  if (dump_) {
+    for (unsigned int i = 0; i < filteredMetaPrimitives.size(); i++) {
+      LogDebug("DTTrigPhase2Prod") << iEvent.id().event() << " filtered mp " << i << ": ";
+      printmP(filteredMetaPrimitives.at(i));
+      LogDebug("DTTrigPhase2Prod");
+    }
+  }
+
+  metaPrimitives.clear();
+  metaPrimitives.erase(metaPrimitives.begin(), metaPrimitives.end());
+
+  if (debug_)
+    LogDebug("DTTrigPhase2Prod") << "DTp2 in event:" << iEvent.id().event() << " we found "
+                                 << filteredMetaPrimitives.size() << " filteredMetaPrimitives (superlayer)"
+                                 << std::endl;
+  if (debug_)
+    LogDebug("DTTrigPhase2Prod") << "filteredMetaPrimitives: starting correlations" << std::endl;
+
+  /////////////////////////////////////
+  //// CORRELATION:
+  /////////////////////////////////////
+  std::vector<metaPrimitive> correlatedMetaPrimitives;
+  if (algo_ == Standard)
+    mpathassociator_->run(iEvent, iEventSetup, dtdigis, filteredMetaPrimitives, correlatedMetaPrimitives);
+  else {
+    for (auto muonpath = outmpaths.begin(); muonpath != outmpaths.end(); ++muonpath) {
+      correlatedMetaPrimitives.emplace_back(metaPrimitive({
```

Comment:
in emplace_back(), just the arguments for the object should be passed; it will automatically call the constructor
(but since metaPrimitive is a struct, its constructor would have to be defined)